### PR TITLE
Add a seed field for the deploy and migration requests [ECR-4366]

### PR DESCRIPTION
**Pull request by aleksuss**

### CODE REVIEW

File: exonum_launcher/launch_state.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ class LaunchState:
     def __init__(self) -> None:
         self._pending_configs: Dict[Configuration, List[str]] = dict()
         self._pending_deployments: Dict[Artifact, List[str]] = dict()
-        self._pending_migrations: Dict[Tuple[str, Artifact], List[str]] = dict()
+        self._pending_migrations: Dict[Tuple[str, Artifact, int], List[str]] = dict()
```
Comment (popzxc) [inline exonum_launcher/launch_state.py:14]: IMHO it makes sense to use a class instead of tuple.

File: exonum_launcher/launcher.py [hunk 2]
```diff
@@ -175,48 +175,52 @@ def unload_all(self) -> None:
...
-            self._explorer.wait_for_txs(tx_hashes)
-            tx_status, description = self._explorer.get_tx_status(tx_hashes[0])
-            if tx_status:
-                self.launch_state.unload_status = ActionResult.Success, description
-            else:
-                self.launch_state.unload_status = ActionResult.Fail, description
 
-        except NotCommittedError as error:
-            self.launch_state.unload_status = ActionResult.Fail, str(error)
+        if tx_hashes:
```
Comment (popzxc) [inline exonum_launcher/launcher.py]: You can use `if not tx_hashes: return` to avoid extra nesting level.

File: exonum_launcher/supervisor.py [hunk 3]
```diff
@@ -71,16 +72,19 @@ def deinitialize(self) -> None:
         """Deinitializes the Supervisor by deinitializing the Protobuf Loader."""
         self._loader.deinitialize()
 
-    def _post_to_supervisor(self, endpoint: str, message: bytes, private: bool = True) -> List[str]:
-        message_data = message.hex()
-        data = json.dumps(message_data)
+    def _post_to_supervisor(self, endpoint: str, message: Optional[Message], private: bool = True) -> List[str]:
```
Comment (popzxc) [inline exonum_launcher/supervisor.py]: Why private method should allow optional `message`? Wouldn't it be right to just not call it on the caller side if there is no message?

File: exonum_launcher/launch_state.py (hunk 1 above)
Comment (aleksuss) [reply exonum_launcher/launch_state.py:14]: Yeah. I've thought about this as well. I think to do this and more related to it in scope of other PR. It requires a massive refactoring and extending test coverage, I suppose. But a lack of time, as you know.

Comment (popzxc) [reply exonum_launcher/launch_state.py:14]: Sure, no problem (:

**Review by popzxc — APPROVED** [review]

**Pull request merged**